src/views/Timeline.vue

```diff
@@ -119,15 +137,31 @@ export default {
             }
           }
         );
-        this.proposals = proposals.timeline;
+        stopLoadingMore.value = response.timeline?.length < loadBy;
+        state.proposals = state.proposals.concat(response.timeline);
       } catch (e) {
         console.log(e);
       }
-      this.loading = false;
     }
-  },
-  async created() {
-    await this.loadProposals();
+
+    // Initialize
+    onMounted(load());
+
+    async function load() {
+      state.loading = true;
+      await loadProposals();
+      state.loading = false;
+    }
+
+    // Change filter
+    function selectState(e) {
+      state.state = e;
+      state.proposals = [];
+      limit.value = loadBy;
+      load();
+    }
+
+    return { state, selectState, loadingMore };
```

Comment:
Why not return { proposals, loading, state etc... } instead of { state ... } ? I feel it's a bit confusing here.

Reply:
Because we're using `reactive()` here instead fo `ref()`

Reply:
It would have to be written `return { proposals: state.proposals }` or I could change it and use `ref()` for storing the reactive state instead.

Reply:
I think it will be more clear to use multiple ref() here. It may make more sense to use reactive to group the vars with same scope but the vars proposals, loading, state, dont have much in common.